### PR TITLE
Add login and fix small bugs

### LINKED ISSUE #56: Fix Transition for Navbar Elements
When you hover over the individual navbar elements, the text size increases but the text color does not change if you hover over the element but not the text itself.

### LINKED ISSUE #57: Add GitHub Login Button
Add login feature that uses the github oauth stuff. Once signed in we should display the user's profile pic.

### PR DESCRIPTION
**Pull request by StephenODea54**
- Added [`zustand`](https://docs.pmnd.rs/zustand/getting-started/introduction) for global state management.
  - This is probably my favorite library for global state management. However, it is bit of a new technology so if y'all oppose I will switch the code to use something like Redux. Speaking of which, Zustand operates a lot like reduce but it has significantly less boilerplate code.
  - Right now:
    1. when the user logs in it will call the `userData` GET route
    2. Assuming the authentication was successful, that data is then saved in our db.
    3. The data is then sent to the frontend, where we store that user's information in our zustand `useUserStore` store (right now I am just keeping the image. We will likely want other stuff like name, is this user an admin? blah blah blah).
    4. I also have the app setup where the access tokens are saved in local storage so that these persist on page refreshes.
    5. You need your github client id (the same one you are using for the backend) to be stored in a `.env` variable called `REACT_APP_GITHUB_CLIENT_ID`
- I have no idea how I did it, but when I was adding the styles it looks like I fixed the navbar links transition issue. I also tweaked the `ul` elements in the navbar to align center;
- Last thing was that I made a small tweak to the tags page. Code was working fine, but the console was yelling at me since none of the tags had a unique ID (all mapped elements in react should have unique identifiers). So I added those in.

Closes #56 
Closes #57

### CODE REVIEW

**Comment by jpatterson933** [comment]
I didn't see it in the code yesterday or in this so maybe I missed it, but are we utilizing the `state` property offered by the github oauth to prevent cross site scripting attacks? I think it would be good practice. If we are and I missed it, just frowny face this comment hehe

File: client/package.json
Comment (jpatterson933) [inline client/package.json:1]: I 100% appreciate the use of new technologies. Shows new, old, doesn't matter, WE CAN USE IT

**Comment by StephenODea54** [comment]
@jpatterson933 i don't have that property listed anywhere in the code. I'll look into it and get back to you.

File: client/src/components/Metrics/index.js
Comment (jpatterson933) [inline client/src/components/Metrics/index.js:1]: This entire page is different and I have it as an issue I am currently working on so really don't need to make any changes to it if you don't want to as this code doesn't exist anymore in the new metrics.

**Review by jpatterson933 — APPROVED** [review]

**Comment by jpatterson933** [comment]
https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps

In first point 1. between `scope` and `allow_signup` is where you can check out the `state` parameter. Again, not a huge deal, but don't want some script kiddie coming in and causing problems we don't need 

Really coming together! Took me a bit to look over everything. I am very impressed by how all of this backend is coming together :)

**Review by jpatterson933 — APPROVED** [review]

**Review by nathanjslim — APPROVED** [review]

**Pull request merged**